### PR TITLE
Add support for interface field in http_response input plugin

### PR DESCRIPTION
**Pull request by GeorgeMac**
This is an attempt to implement the `interface` field in the `http_response` input plugin.

The `interface` field can be populated with an interface name such as `eth0`. A local address associated with the network interface will be derived and used for dialing.

Resolves: https://github.com/influxdata/telegraf/issues/5856

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/http_response/http_response_test.go [hunk 1]
```diff
@@ -210,6 +212,69 @@ func TestFields(t *testing.T) {
...
+	var (
+		mux = setUpTestMux()
+		ts  = httptest.NewServer(mux)
+	)
+
+	defer ts.Close()
+
+	intf, err := findInterface()
+	if err != nil {
+		t.Fatal(err)
```
Comment (glinton) [inline plugins/inputs/http_response/http_response_test.go]: I likely would have just `t.Skip(err)`, but then again I believe `httptest.NewServer` requires at least the loopback interface, so this is fine.

Comment (GeorgeMac) [reply plugins/inputs/http_response/http_response_test.go]: I can change that if skip is preferred. I didn’t realize skip existed in that form.

Comment (GeorgeMac) [reply plugins/inputs/http_response/http_response_test.go]: I can change that if skip is preferred. I didn’t realize skip existed in that form.

File: plugins/inputs/http_response/http_response_test.go [hunk 2]
```diff
@@ -210,6 +212,69 @@ func TestFields(t *testing.T) {
...
+
+func TestInterface(t *testing.T) {
+	var (
+		mux = setUpTestMux()
+		ts  = httptest.NewServer(mux)
+	)
+
+	defer ts.Close()
+
+	intf, err := findInterface()
```
[the remark this answers is not in the document]
Comment (GeorgeMac) [reply plugins/inputs/http_response/http_response_test.go]: Absolutely. That makes more sense. I was trying to test a non-loopback interface, which in hindsight makes no sense. I will make it get the name of the first loopback interface it finds.

**Pull request merged**